<?xml version="1.0"?>
<!-- This URDF was automatically created by SolidWorks to URDF Exporter! Originally created by Stephen Brawner (brawner@gmail.com) 
     Commit Version: 1.6.0-4-g7f85cfe  Build Version: 1.6.7995.38578
     For more information, please see http://wiki.ros.org/sw_urdf_exporter -->
<robot  name="conveyor_belt" xmlns:xacro="http://ros.org/wiki/xacro">


  <xacro:macro name="body_color">
    <visual>
      <material>
        <ambient>0.501960784313725 0.501960784313725 0.501960784313725 1</ambient>
        <diffuse>0.501960784313725 0.501960784313725 0.501960784313725 1</diffuse>
        <specular>0.1 0.1 0.1 1</specular>
        <emissive>0 0 0 0</emissive>
      </material>
    </visual>
  </xacro:macro>
  <xacro:macro name="belt_color">
    <visual>
      <material>
        <ambient>0 0 0 1</ambient>
        <diffuse>0 0 0  1</diffuse>
        <specular>0.1 0.1 0.1 1</specular>
        <emissive>0 0 0 0</emissive>
      </material>
    </visual>
  </xacro:macro>


  <link
    name="base_link">
    <inertial>
      <origin
        xyz="1.15972439062802E-17 -0.0273162100367329 1.02144955943776E-17"
        rpy="0 0 0" />
      <mass
        value="2430.68583470577" />
      <inertia
        ixx="431.198140853202"
        ixy="2.18658245479853E-14"
        ixz="4.38017677684144E-14"
        iyy="7514.31711977665"
        iyz="3.72364768160213E-15"
        izz="7545.20095484556" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://conveyor_belt/meshes/base_link.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.501960784313725 0.501960784313725 0.501960784313725 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://conveyor_belt/meshes/base_link.STL" />
      </geometry>
    </collision>
  </link>
  <link
    name="belt_link">
    <inertial>
      <origin
        xyz="2.84 0.00499999999999995 2.76334119905601E-17"
        rpy="0 0 0" />
      <mass
        value="56.8" />
      <inertia
        ixx="4.73380666666667"
        ixy="3.71142749699624E-34"
        ixz="2.20830761084774E-15"
        iyy="157.442026666667"
        iyz="2.30486925841449E-19"
        izz="152.709166666667" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://conveyor_belt/meshes/belt_link.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0 0 0 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://conveyor_belt/meshes/belt_link.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="belt_joint"
    type="prismatic">
    <origin
      xyz="-2.83 0 0.3"
      rpy="1.5708 0 0" />
    <parent
      link="base_link" />
    <child
      link="belt_link" />
    <axis
      xyz="-1 0 0" />
    <limit
      lower="0"
      upper="0"
      effort="0"
      velocity="0" />
  </joint>


<!-- Gazebo Colors -->
  <gazebo reference="base_link">
    <xacro:body_color />
  </gazebo>
  <gazebo reference="belt_link">
    <xacro:belt_color />
  </gazebo>
  

<!-- Custom Gazebo Plugin for Conveyor Belt  -->
  <gazebo>
  <plugin filename="libconveyor_belt_plugin.so" name="conveyor_belt_plugin">
        <ros>
          <namespace></namespace>
        </ros>
        <max_velocity>0.2</max_velocity>
        <publish_rate>10</publish_rate>
  </plugin>
  </gazebo>

</robot>